llvm/lib/Target/RISCV/RISCVFeatures.td

```diff
@@ -22,11 +22,20 @@
 //             uses the record name by replacing Feature with Has.
 // value     - Value to assign to the field in RISCVSubtarget when this
 //             extension is enabled. Usually "true", but can be changed.
-class RISCVExtension<string name, int major, int minor, string desc,
+// IsExperimental
+//           - Indicates whether the extension is experimental
+class RISCVExtension<int major, int minor, string desc,
                      list<SubtargetFeature> implies = [],
                      string fieldname = !subst("Feature", "Has", NAME),
-                     string value = "true">
-    : SubtargetFeature<name, fieldname, value, desc, implies> {
+                     string value = "true", bit IsExperimental = false>
+    : SubtargetFeature<
+        !if(IsExperimental, "experimental-", "")
```

Comment:
You can pass `""` here and use `let` to override the `Name` field, so that we can remove duplicated `!subst`s via using `defvar` statements.

Reply:
And, we have some naming conventions for extension features after this change, so please document above that the name must start with `FeatureStdExt` for standard extensions and `FeatureVendor` for vendor extensions.

Reply:
Nice!